Makefile

```diff
@@ -17,12 +17,15 @@ all: manager
 bin/kubebuilder:
 	hack/download-kubebuilder-local.sh
 
-# Run tests
+# Run unit tests
 test: generate fmt vet manifests bin/kubebuilder
 	KUBEBUILDER_ASSETS="$(shell pwd)/bin/kubebuilder/bin" go test ./... -coverprofile cover.out
 
+# Run end to end tests in a local Kind cluster. We do not clean up after running the tests to
+#  a) speed up the test run time slightly
+#  b) allow debug sessions to be attached to figure out what caused failures
 kind: generate fmt vet manifests
-	go test ./internal/test/e2e --kind --repo-root ${CURDIR} -v
+	go test ./internal/test/e2e --kind --repo-root ${CURDIR} -v --cleanup=false
```

Comment:
I'm a bit dubious about this `--cleanup=false` business. Won't it mean that running `make kind` twice in a row will fail unless you destroy the KIND cluster yourself between the runs?

Reply:
happy to revert that, but for context it cuts about 20-30% from the build time